packages/expo-system-ui/android/src/main/java/expo/modules/systemui/SystemUIModule.kt

```diff
@@ -0,0 +1,262 @@
+        val style = if (controller.isAppearanceLightStatusBars) "light" else "dark"
+        promise.resolve(style)
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun setStatusBarForegroundStyle(style: String, promise: Promise) {
+    activity.runOnUiThread {
+      WindowInsetsControllerCompat(activity.window, activity.window.decorView).let { controller ->
+        when (style) {
+          "light" -> controller.isAppearanceLightStatusBars = false
+          "dark" -> controller.isAppearanceLightStatusBars = true
+          else -> {
+            promise.reject("invalid-value", "Value \"$style\" is not a valid Status Bar foreground style!")
+            return@let
+          }
+        }
+        promise.resolve(null)
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun getNavigationBarForegroundStyle(promise: Promise) {
+    activity.runOnUiThread {
+      WindowInsetsControllerCompat(activity.window, activity.window.decorView).let { controller ->
+        val style = if (controller.isAppearanceLightNavigationBars) "light" else "dark"
+        promise.resolve(style)
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun setNavigationBarForegroundStyle(style: String, promise: Promise) {
+    activity.runOnUiThread {
+      WindowInsetsControllerCompat(activity.window, activity.window.decorView).let { controller ->
+        when (style) {
+          "light" -> controller.isAppearanceLightNavigationBars = false
+          "dark" -> controller.isAppearanceLightNavigationBars = true
+          else -> {
+            promise.reject("invalid-value", "Value \"$style\" is not a valid Navigation Bar foreground style!")
+            return@let
+          }
+        }
+        promise.resolve(null)
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun setNavigationBarVisibility(visibility: String, promise: Promise) {
+    activity.runOnUiThread {
+      WindowInsetsControllerCompat(activity.window, activity.window.decorView).let { controller ->
+        when (visibility) {
+          "visible" -> {
+            controller.show(WindowInsetsCompat.Type.navigationBars())
+          }
+          "hidden" -> {
+            controller.hide(WindowInsetsCompat.Type.navigationBars())
+            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
+          }
+        }
+      }
+      promise.resolve(null)
+    }
+  }
+
+  @ExpoMethod
+  fun setStatusBarVisibility(visibility: String, promise: Promise) {
+    activity.runOnUiThread {
+      WindowInsetsControllerCompat(activity.window, activity.window.decorView).let { controller ->
+        when (visibility) {
+          "visible" -> {
+            controller.show(WindowInsetsCompat.Type.statusBars())
+          }
+          "hidden" -> {
+            controller.hide(WindowInsetsCompat.Type.statusBars())
+            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
```

Comment:
Maybe this shouldn't be here since it appears to also hide the navigation bar. The `systemBarsBehavior` looks like it can be called from another method anyways.